src/Site.js

```diff
@@ -435,6 +456,140 @@ Site.prototype.createPage = function (config) {
+  const indexPagePath = path.join(this.rootPath, INDEX_MARKDOWN_FILE);
+  const fileNames = ['README.md', 'Home.md'];
+  const filePath = fileNames.find(fileName => fs.existsSync(path.join(this.rootPath, fileName)));
+  return fs.copyAsync(path.join(this.rootPath, filePath), indexPagePath)
+    .catch(() => Promise.reject(new Error(`Failed to copy over ${filePath}`)));
+};
+
+/**
+ * Adds an about page to site if not present.
+ */
+Site.prototype.addAboutPage = function () {
+  const aboutPath = path.join(this.rootPath, ABOUT_MARKDOWN_FILE);
+  return fs.accessAsync(aboutPath)
+    .catch(() => {
+      if (fs.existsSync(aboutPath)) {
+        return Promise.resolve();
+      }
+      return fs.outputFileAsync(aboutPath, ABOUT_MARKDOWN_DEFAULT);
+    });
+};
+
+/**
+ * Adds top navigation menu to default layout of site.
+ */
+Site.prototype.addTopNavToDefaultLayout = function () {
+  const siteLayoutPath = path.join(this.rootPath, LAYOUT_FOLDER_PATH);
+  const siteLayoutHeaderDefaultPath = path.join(siteLayoutPath, LAYOUT_DEFAULT_NAME, 'header.md');
+
+  return fs.outputFileAsync(siteLayoutHeaderDefaultPath, TOP_NAV_DEFAULT);
+};
+
+/**
+ * Adds a footer to default layout of site.
+ */
+Site.prototype.addFooterToDefaultLayout = function () {
+  const footerPath = path.join(this.rootPath, FOOTER_PATH);
+  const siteLayoutPath = path.join(this.rootPath, LAYOUT_FOLDER_PATH);
+  const siteLayoutFooterDefaultPath = path.join(siteLayoutPath, LAYOUT_DEFAULT_NAME, 'footer.md');
+  const wikiFooterPath = path.join(this.rootPath, WIKI_FOOTER_PATH);
+
+  return fs.accessAsync(wikiFooterPath)
+    .then(() => {
+      const footerContent = fs.readFileSync(wikiFooterPath, 'utf8');
+      const wrappedFooterContent = `<footer>\n\t${footerContent}\n</footer>`;
+      return fs.outputFileAsync(siteLayoutFooterDefaultPath, wrappedFooterContent);
+    })
+    .catch(() => {
+      if (fs.existsSync(footerPath)) {
+        return fs.copyAsync(footerPath, siteLayoutFooterDefaultPath);
+      }
+      return Promise.resolve();
+    });
+};
+
+/**
+ * Adds a site navigation bar to the default layout of the site.
+ */
+Site.prototype.addSiteNavToDefaultLayout = function () {
+  const siteLayoutPath = path.join(this.rootPath, LAYOUT_FOLDER_PATH);
+  const siteLayoutSiteNavDefaultPath = path.join(siteLayoutPath, LAYOUT_DEFAULT_NAME, 'navigation.md');
+  const wikiSiteNavPath = path.join(this.rootPath, WIKI_SITE_NAV_PATH);
+
+  return fs.accessAsync(wikiSiteNavPath)
+    .then(() => {
+      const siteNavContent = fs.readFileSync(wikiSiteNavPath, 'utf8');
+      const wrappedSiteNavContent = `<navigation>\n${siteNavContent}\n</navigation>`;
+      logger.info(`Copied over the existing _Sidebar.md file to ${path.relative(this.rootPath,
+                                                                                siteLayoutSiteNavDefaultPath)
+      }\n\tCheck https://markbind.org/userGuide/tweakingThePageStructure.html#site-navigation-menus`
```

Comment:
The `\t` here mis-aligns the message. While you could add 6 spaces, I think its not very clean. I'm fine with logging another info line directly after.

Reply:
I'll log the info line directly after.